Make distclean fails when --enable-tests are configured

``` text
rm -f TAGS ID GTAGS GRTAGS GSYMS GPATH tags
rm -f ./.deps/dosbox.Po
rm -f Makefile
Making distclean in tests
Making distclean in ../src/misc
make[2]: *** No rule to make target 'distclean'.  Stop.
make[1]: *** [Makefile:504: distclean-recursive] Error 1
make: *** [Makefile:402: distclean-recursive] Error 1
```

This leaves behind a mix of autotools generated outputs (such as `./configure`), which then fools the `build.sh` into thinking we're all set to run, so that in turn fails:

``` text
./scripts/build.sh -t debug -c gcc

Compiling a debug build using gcc on Linux-x86_64
=================================================

Environment
-----------
[gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0]
AR="gcc-ar"
CC="ccache gcc"
CXX="ccache g++"
LD="gcc"
RANLIB="gcc-ranlib"
ARFLAGS="Dcr"
CFLAGS="-Wall -pipe -fdiagnostics-color=auto -fpch-preprocess -g -fstack-protector -fno-omit-frame-pointer"
CXXONLY="-Weffc++"
LDFLAGS="-Wl,--as-needed"
LIBS=""
export AR CC CXX LD RANLIB ARFLAGS CFLAGS CXXONLY LDFLAGS LIBS CXXFLAGS="${CFLAGS} ${CXXONLY}"

Launching
---------
make --no-print-directory all-recursive
Making all in docs
make[2]: *** No rule to make target 'all'.  Stop.
make[1]: *** [Makefile:402: all-recursive] Error 1
make: *** [Makefile:343: all] Error 2
```

I tried added a `distclean-local:` target to `tests`, and then `docs`, and then `src/misc` -- but that's as far as I got before giving up. The tests's makefile.am looks fine, so I'm not sure what sent this off the rails.

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -15,7 +15,7 @@ jobs:
             max_warnings: 316
           - name: MSVC 64-bit
             arch: x64
-            max_warnings: 2041
+            max_warnings: 2035
     steps:
       - name:  Backup existing vcpkg installation
         shell: pwsh

--- a/Makefile.am
+++ b/Makefile.am
@@ -17,14 +17,18 @@ SUBDIRS = docs include src tests
 # (note: with recursive makefiles, these files could be defined as individual distclean-local:
 # targets inside each sub-directory's Makefile.am, however listing them in one target here
 # is easier manage as they can all be seen at once.)
+#
+# This results in 'make distclean' working the same as 'git clean -fdx', even when code
+# was distributed via a tarball, and not cloned from a Git repo.
 
 distclean-local:
+	-rm ar-lib
 	-rm aclocal.m4
 	-rm build.log
 	-rm compile
+	-rm compile_commands.json
 	-rm config.guess
-	-rm config.h.in
-	-rm config.h.in~
+	-rm config.h.in*
 	-rm config.log
 	-rm config.sub
 	-rm configure

--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -19,4 +19,11 @@ tests_SOURCES = \
 
 tests_LDADD = ../src/misc/libmisc.a
 
+# Override automake's distclean target to prevent it recursing into
+# SUBDIRS and failing (it was already invoked in there via src/Makefile.am).
+distclean:
+	-rm tests$(EXEEXT)
+	-rm *.o
+	-rm Makefile
+
 endif

